src/store/sagas/stats.ts

```diff
@@ -0,0 +1,65 @@
+import { IStats, action } from '@reducers/stats'
+import { call, put, all, spawn, takeEvery, select } from 'typed-redux-saga'
+import { network } from '@selectors/solanaConnection'
+import axios from 'axios'
+
+export function* getCurrentNetwork() {
+  const currentNetwork = yield* select(network)
+  return currentNetwork.toLowerCase()
+}
+const getData = async (name: string) => {
+  return await axios.get(`https://api.synthetify.io/stats/${name}`)
+}
+
+export function* fetchData() {
+  const dataTmp: IStats = {
+    value: [],
+    last24: { volume: 0, collateral: 0, mint: 0, debt: 0, fee: 0 }
+  }
+  try {
+    const currentNetwork = yield* call(getCurrentNetwork)
+    const response = yield* call(getData, currentNetwork)
+    const elements = ['volume', 'mint', 'burn', 'liquidation', 'userCount']
+    elements.forEach(element => {
+      fillData(response.data, element, dataTmp)
```

Comment:
instead of calling this function 5 times with the same responsed.data, we can just call it once and assign value in all the arrays in one loop